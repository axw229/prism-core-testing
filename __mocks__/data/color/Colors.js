// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
