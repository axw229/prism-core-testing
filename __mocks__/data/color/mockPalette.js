const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
