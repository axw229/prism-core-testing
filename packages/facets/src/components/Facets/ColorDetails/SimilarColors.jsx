// @flow
import React, { useContext } from 'react'
import { useSelector } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import ConfigurationContext, { type ConfigurationContextType } from 'src/contexts/ConfigurationContext/ConfigurationContext'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import 'src/scss/convenience/visually-hidden.scss'

type Props = { color: Color, onColorChanged: Color => void }
const SimilarColors = ({ color, onColorChanged }: Props) => {
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  return (
    <>
      <h5 className='visually-hidden'><FormattedMessage id='SIMILAR_COLORS' /></h5>
      <ul className='color-info__similar-colors'>
        {color.similarColors.map(colorId => colorMap[colorId]).map((color: Color) => {
          return (color &&
            <li key={color.colorNumber} className='color-info__similar-color'>
              <button
                className='color-info__similar-color-info'
                style={{ backgroundColor: color.hex }}
                onClick={() => {
                  onColorChanged(color)
                  GA.event({ category: 'Color Detail / Similar Color', action: 'View Similar Color', label: color.name }, GA_TRACKER_NAME_BRAND[brandId])
                }}
                aria-label={`${color.brandKey} ${color.colorNumber} ${color.name}`}
              />
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default SimilarColors
