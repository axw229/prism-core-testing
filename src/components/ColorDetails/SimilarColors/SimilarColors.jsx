// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { find } from 'lodash'

import { paintAllSceneSurfaces } from '../../../actions/scenes'

type Props = {
  colors: Array,
  color: Object,
  paintAllSceneSurfaces: Function
}

class SimilarColors extends PureComponent<Props> {
  render () {
    const similarColors = this.similarColors()

    return (
      <React.Fragment>
        <strong>Similar Colors</strong>
        <hr />
        <ul>
          {similarColors.map(color => {
            return (
              <li key={color.id}><button onClick={() => this.selectColor(color)}>{color.name}</button></li>
            )
          })}
        </ul>
        <hr />
      </React.Fragment>
    )
  }

  similarColors () {
    const { colors, color } = this.props
    const similarColorIds = color.similarColors || []

    const similarColors = similarColorIds.map(similarColorId => find(colors, color => color.id === similarColorId))

    return similarColors
  }

  selectColor (color) {
    this.props.paintAllSceneSurfaces(color)
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    paintAllSceneSurfaces: (color) => {
      dispatch(paintAllSceneSurfaces(color))
    }
  }
}

export default connect(null, mapDispatchToProps)(SimilarColors)
