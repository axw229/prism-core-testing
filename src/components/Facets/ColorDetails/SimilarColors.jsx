// @flow
import React from 'react'
import { useSelector } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import * as GA from 'src/analytics/GoogleAnalytics'
import 'src/scss/convenience/visually-hidden.scss'

type Props = { color: Color, onColorChanged: Color => void }
const SimilarColors = ({ color, onColorChanged }: Props) => {
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  return (
    <>
      <h5 className='visually-hidden'><FormattedMessage id='SIMILAR_COLORS' /></h5>
      <ul className='color-info__similar-colors'>
        {color.similarColors.map(colorId => colorMap[colorId]).map((color: Color) => {
          return (color &&
            <li key={color.colorNumber} className='color-info__similar-color'>
              <button
                className='color-info__similar-color-info'
                style={{ backgroundColor: color.hex }}
                onClick={() => {
                  onColorChanged(color)
                  GA.event({ category: 'Color Detail / Similar Color', action: 'View Similar Color', label: color.name })
                }}
                aria-label={`${color.brandKey} ${color.colorNumber} ${color.name}`}
              />
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default SimilarColors
