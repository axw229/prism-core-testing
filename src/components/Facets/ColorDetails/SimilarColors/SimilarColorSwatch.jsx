// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import * as GA from 'src/analytics/GoogleAnalytics'

import type { Color } from '../../../../shared/types/Colors'

import { generateColorDetailsPageUrl } from '../../../../shared/helpers/ColorUtils'

type Props = {
  color: Color
}

function SimilarColorSwatch ({ color }: Props) {
  const BASE_CLASS = 'color-info'
  const handleClick = () => {
    GA.event({
      category: 'Color Detail / Similar Color',
      action: 'View Similar Color',
      label: color.name
    })
  }

  return (
    <li className={`${BASE_CLASS}__similar-color`} style={{ backgroundColor: color.hex }}>
      <Link to={generateColorDetailsPageUrl(color)} className={`${BASE_CLASS}__similar-color-info`} onClick={handleClick} style={{ backgroundColor: color.hex }}>
        <div className={`${BASE_CLASS}__similar-color-info-wrapper`}>
          <span className={`${BASE_CLASS}__similar-color-brand-key`} >{color.brandKey}</span> <span className={`${BASE_CLASS}__similar-color-number`} >{color.colorNumber}</span>
          <p className={`${BASE_CLASS}__similar-color-name`} >{color.name}</p>
        </div>
      </Link>
    </li>
  )
}

export default SimilarColorSwatch
