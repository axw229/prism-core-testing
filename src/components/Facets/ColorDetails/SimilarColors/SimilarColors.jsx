// @flow
import React, { PureComponent } from 'react'
import { FormattedMessage } from 'react-intl'

import type { Color, ColorMap } from '../../../../shared/types/Colors'

import SimilarColorSwatch from './SimilarColorSwatch'

type Props = {
  colors: ColorMap,
  color: Color
}

type State = {
  activeColorID: number | void
}

class SimilarColors extends PureComponent<Props, State> {
  static baseClass = 'color-info'

  constructor (props: Props) {
    super(props)

    this.state = { activeColorID: void (0) }
  }

  render () {
    const { colors, color } = this.props
    const similarColors = color.similarColors

    return (
      <React.Fragment>
        <h5 className='visually-hidden'><FormattedMessage id='SIMILAR_COLORS' /></h5>
        <ul className={`${SimilarColors.baseClass}__similar-colors`}>
          {similarColors.map((colorId: string) => {
            const color = colors[colorId]
            if (color) {
              return <SimilarColorSwatch key={colorId} color={color} />
            }
          }).filter(color => !!color)}
        </ul>
        <hr />
      </React.Fragment>
    )
  }
}

export default SimilarColors
