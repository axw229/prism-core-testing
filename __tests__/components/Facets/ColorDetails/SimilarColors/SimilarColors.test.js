/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import SimilarColors from 'src/components/Facets/ColorDetails/SimilarColors/SimilarColors'
import * as Colors from '__mocks__/data/color/Colors'
import { FormattedMessage } from 'react-intl'

const color = Colors.getColor()
const colors = Colors.getAllColors()

const getSimilarColors = (props) => {
  return shallow(<SimilarColors {...props} />)
}

describe('SimilarColors component with props', () => {
  let similarColors
  beforeEach(() => {
    if (!similarColors) {
      similarColors = getSimilarColors({ colors: colors, color: color })
    }
  })

  it('SimilarColors is rendering', () => {
    expect(similarColors.exists()).toBe(true)
  })

  it('SimilarColors is rendering H5 with FormattedMessage', () => {
    expect(similarColors.find('h5.visually-hidden').contains(<FormattedMessage id='SIMILAR_COLORS' />)).toBeTruthy()
  })

  it('SimilarColors is rendering SimilarColorSwatch if color object contains similarColors', () => {
    if (color.similarColors) {
      expect(similarColors.find('SimilarColorSwatch').exists()).toBe(true)
    }
  })
})
