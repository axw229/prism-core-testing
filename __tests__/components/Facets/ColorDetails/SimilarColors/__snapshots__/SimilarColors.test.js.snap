// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarColors component with props SimilarColors Snapshot with props 1`] = `
<Fragment>
  <h5
    className="visually-hidden"
  >
    <FormattedMessage
      id="SIMILAR_COLORS"
      values={Object {}}
    />
  </h5>
  <ul
    className="color-info__similar-colors"
  >
    <SimilarColorSwatch
      color={
        Object {
          "blue": 221,
          "brandKey": "SW",
          "brandedCollectionNames": Array [
            "Pottery Barn Fall/Winter 2018",
            "Pottery Barn Fall/Winter 2018",
          ],
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6568",
          "coordinatingColors": Object {
            "coord1ColorId": "2687",
            "coord2ColorId": "11244",
            "whiteColorId": "2681",
          },
          "description": Array [
            "Subdued",
            "Muted",
            "Brilliant",
            "Radiant",
          ],
          "green": 213,
          "hex": "#edd5dd",
          "hue": 0.9444444444444445,
          "id": "2255",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8823529411764706,
          "lrv": 70.758,
          "name": "Lighthearted Pink",
          "red": 237,
          "saturation": 0.3999999999999999,
          "similarColors": Array [
            "2248",
            "2262",
            "2269",
            "2276",
            "1974",
            "1981",
            "2658",
            "1995",
            "2662",
            "1988",
          ],
          "storeStripLocator": "103-C1",
        }
      }
      key="2255"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 218,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6575",
          "coordinatingColors": Object {
            "coord1ColorId": "2952",
            "coord2ColorId": "2755",
            "whiteColorId": "2681",
          },
          "description": Array [
            "Soft",
            "Brilliant",
            "Radiant",
          ],
          "green": 211,
          "hex": "#f1d3da",
          "hue": 0.9611111111111111,
          "id": "2262",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8862745098039215,
          "lrv": 70.62,
          "name": "Priscilla",
          "red": 241,
          "saturation": 0.5172413793103448,
          "similarColors": Array [
            "2269",
            "2248",
            "2255",
            "2276",
            "2543",
            "2535",
            "1995",
            "1988",
            "2662",
            "1981",
          ],
          "storeStripLocator": "104-C1",
        }
      }
      key="2262"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 215,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6582",
          "coordinatingColors": Object {
            "coord1ColorId": "2686",
            "coord2ColorId": "2416",
            "whiteColorId": "2681",
          },
          "description": Array [
            "Soft",
            "Brilliant",
            "Radiant",
          ],
          "green": 210,
          "hex": "#f1d2d7",
          "hue": 0.9731182795698924,
          "id": "2269",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.884313725490196,
          "lrv": 69.699,
          "name": "Impatiens Petal",
          "red": 241,
          "saturation": 0.5254237288135593,
          "similarColors": Array [
            "2262",
            "2276",
            "2248",
            "2255",
            "2543",
            "1995",
            "2535",
            "1988",
            "2002",
            "1981",
          ],
          "storeStripLocator": "105-C1",
        }
      }
      key="2269"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 221,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "SW6976_COLOR_FAMILY_NAME",
          ],
          "colorNumber": "6976",
          "coordinatingColors": Object {
            "coord1ColorId": "1628",
            "coord2ColorId": "2071",
          },
          "description": Array [
            "Subdued",
            "Muted",
            "Brilliant",
            "Radiant",
          ],
          "green": 204,
          "hex": "#e6ccdd",
          "hue": 0.891025641025641,
          "id": "2662",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8509803921568628,
          "lrv": 65.078,
          "name": "Vanity Pink",
          "red": 230,
          "saturation": 0.34210526315789486,
          "similarColors": Array [
            "2531",
            "2248",
            "2522",
            "2535",
            "2249",
            "2658",
            "2255",
            "2262",
            "2256",
            "2539",
          ],
        }
      }
      key="2662"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 213,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "SW6848_COLOR_FAMILY_NAME",
          ],
          "colorNumber": "6848",
          "coordinatingColors": Object {
            "coord1ColorId": "2583",
            "coord2ColorId": "2867",
          },
          "description": Array [
            "Soft",
            "Brilliant",
            "Radiant",
          ],
          "green": 201,
          "hex": "#edc9d5",
          "hue": 0.9444444444444443,
          "id": "2535",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8588235294117648,
          "lrv": 64.433,
          "name": "Panache Pink",
          "red": 237,
          "saturation": 0.5000000000000004,
          "similarColors": Array [
            "2543",
            "2539",
            "2531",
            "2256",
            "2249",
            "2262",
            "2263",
            "2248",
            "2269",
            "2662",
          ],
        }
      }
      key="2535"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 215,
          "brandKey": "SW",
          "brandedCollectionNames": Array [
            "Kids' Colors",
          ],
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6589",
          "coordinatingColors": Object {
            "coord1ColorId": "2765",
            "coord2ColorId": "11262",
            "whiteColorId": "2681",
          },
          "description": Array [
            "Soft",
            "Brilliant",
            "Radiant",
          ],
          "green": 213,
          "hex": "#f2d5d7",
          "hue": 0.9885057471264368,
          "id": "2276",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.892156862745098,
          "lrv": 71.119,
          "name": "Alyssum",
          "red": 242,
          "saturation": 0.5272727272727267,
          "similarColors": Array [
            "2269",
            "1995",
            "2262",
            "1988",
            "2002",
            "2255",
            "1981",
            "2248",
            "2774",
            "2275",
          ],
          "storeStripLocator": "106-C1",
        }
      }
      key="2276"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 226,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "SW6972_COLOR_FAMILY_NAME",
          ],
          "colorNumber": "6972",
          "coordinatingColors": Object {
            "coord1ColorId": "1943",
            "coord2ColorId": "2825",
          },
          "description": Array [
            "Diluted",
            "Wan",
            "Brilliant",
            "Radiant",
          ],
          "green": 212,
          "hex": "#e4d4e2",
          "hue": 0.854166666666667,
          "id": "2658",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8627450980392157,
          "lrv": 69.199,
          "name": "Joyful Lilac",
          "red": 228,
          "saturation": 0.22857142857142865,
          "similarColors": Array [
            "2521",
            "2255",
            "2522",
            "2248",
            "2662",
            "2242",
            "2507",
            "1967",
            "1974",
            "2515",
          ],
        }
      }
      key="2658"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 209,
          "brandKey": "SW",
          "brandedCollectionNames": Array [
            "Pottery Barn Fall/Winter 2018",
            "Pottery Barn Fall/Winter 2018",
          ],
          "colorFamilyNames": Array [
            "Red",
          ],
          "colorNumber": "6288",
          "coordinatingColors": Object {
            "coord1ColorId": "1975",
            "coord2ColorId": "11256",
            "whiteColorId": "2681",
          },
          "description": Array [
            "Diluted",
            "Wan",
            "Brilliant",
            "Radiant",
          ],
          "green": 205,
          "hex": "#e0cdd1",
          "hue": 0.9649122807017544,
          "id": "1974",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8411764705882353,
          "lrv": 64.293,
          "name": "Rosebud",
          "red": 224,
          "saturation": 0.23456790123456778,
          "similarColors": Array [
            "1981",
            "1967",
            "2255",
            "1995",
            "1988",
            "2248",
            "2276",
            "2275",
            "2002",
            "2269",
          ],
          "storeStripLocator": "109-C1",
        }
      }
      key="1974"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 211,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "SW6856_COLOR_FAMILY_NAME",
          ],
          "colorNumber": "6856",
          "coordinatingColors": Object {
            "coord1ColorId": "1691",
            "coord2ColorId": "1714",
          },
          "description": Array [
            "Soft",
            "Brilliant",
            "Radiant",
          ],
          "green": 202,
          "hex": "#f1cad3",
          "hue": 0.9615384615384616,
          "id": "2543",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8686274509803922,
          "lrv": 65.596,
          "name": "Reverie Pink",
          "red": 241,
          "saturation": 0.5820895522388061,
          "similarColors": Array [
            "2535",
            "2539",
            "2263",
            "2256",
            "2269",
            "2262",
            "2277",
            "2249",
            "2531",
            "2248",
          ],
        }
      }
      key="2543"
    />
    <SimilarColorSwatch
      color={
        Object {
          "blue": 218,
          "brandKey": "SW",
          "colorFamilyNames": Array [
            "SW6844_COLOR_FAMILY_NAME",
          ],
          "colorNumber": "6844",
          "coordinatingColors": Object {
            "coord1ColorId": "2885",
            "coord2ColorId": "2883",
          },
          "description": Array [
            "Fairly Colorful",
            "Moderately Colorful",
            "Brilliant",
            "Radiant",
          ],
          "green": 200,
          "hex": "#eac8da",
          "hue": 0.9117647058823529,
          "id": "2531",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.8509803921568627,
          "lrv": 63.577,
          "name": "Merry Pink",
          "red": 234,
          "saturation": 0.4473684210526314,
          "similarColors": Array [
            "2249",
            "2535",
            "2662",
            "2256",
            "2539",
            "2543",
            "2532",
            "2248",
            "2263",
            "2262",
          ],
        }
      }
      key="2531"
    />
  </ul>
</Fragment>
`;
